azurerm/internal/services/network/subnet_service_endpoint_storage_policy_resource.go

```diff
@@ -0,0 +1,258 @@
+			}
+		}
+
+		if resp.ID != nil && *resp.ID != "" {
+			return tf.ImportAsExistsError("azurerm_subnet_service_endpoint_storage_policy", *resp.ID)
+		}
+	}
+
+	param := network.ServiceEndpointPolicy{
+		Location: &location,
+		ServiceEndpointPolicyPropertiesFormat: &network.ServiceEndpointPolicyPropertiesFormat{
+			ServiceEndpointPolicyDefinitions: expandServiceEndpointPolicyDefinitions(d.Get("definition").([]interface{})),
+		},
+		Tags: tags.Expand(d.Get("tags").(map[string]interface{})),
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resourceGroup, name, param)
+	if err != nil {
+		return fmt.Errorf("creating Subnet Service Endpoint Storage Policy %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting for creation of Subnet Service Endpoint Storage Policy %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, name, "")
+	if err != nil {
+		return fmt.Errorf("retrieving Subnet Service Endpoint Storage Policy %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if resp.ID == nil || *resp.ID == "" {
+		return fmt.Errorf("empty or nil ID returned for Subnet Service Endpoint Storage Policy %q (Resource Group %q) ID", name, resourceGroup)
+	}
+
+	id, err := parse.SubnetServiceEndpointStoragePolicyID(*resp.ID)
+	if err != nil {
+		return err
+	}
+	d.SetId(id.ID(subscriptionId))
+
+	return resourceArmSubnetServiceEndpointStoragePolicyRead(d, meta)
+}
+
+func resourceArmSubnetServiceEndpointStoragePolicyRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Network.ServiceEndpointPoliciesClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.SubnetServiceEndpointStoragePolicyID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.ServiceEndpointPolicyName, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] Subnet Service Endpoint Storage Policy %q was not found in Resource Group %q - removing from state!", id.ServiceEndpointPolicyName, id.ResourceGroup)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("retrieving Subnet Service Endpoint Storage Policy %q (Resource Group %q): %+v", id.ServiceEndpointPolicyName, id.ResourceGroup, err)
+	}
+
+	d.Set("name", id.ServiceEndpointPolicyName)
+	d.Set("resource_group_name", id.ResourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+	if prop := resp.ServiceEndpointPolicyPropertiesFormat; prop != nil {
+		if err := d.Set("definition", flattenServiceEndpointPolicyDefinitions(prop.ServiceEndpointPolicyDefinitions)); err != nil {
+			return fmt.Errorf(`setting "definition": %v`, err)
```

Comment:
_minor_ but the rest of the provider uses this "``" rather than `""`:

```suggestion
			return fmt.Errorf("setting `definition`: %v", err)
```